libnavui-androidauto/api/current.txt

```diff
@@ -341,7 +341,7 @@ package com.mapbox.androidauto.navigation {
 
   public final class CarNavigationInfoMapper {
     ctor public CarNavigationInfoMapper(android.content.Context context, com.mapbox.androidauto.navigation.maneuver.CarManeuverInstructionRenderer carManeuverInstructionRenderer, com.mapbox.androidauto.navigation.maneuver.CarManeuverIconRenderer carManeuverIconRenderer, com.mapbox.androidauto.navigation.lanes.CarLanesImageRenderer carLanesImageGenerator);
-    method public androidx.car.app.navigation.model.NavigationTemplate.NavigationInfo? mapNavigationInfo(com.mapbox.bindgen.Expected<com.mapbox.navigation.ui.maneuver.model.ManeuverError,java.util.List<com.mapbox.navigation.ui.maneuver.model.Maneuver>> expectedManeuvers, java.util.List<? extends com.mapbox.navigation.ui.shield.model.RouteShield> routeShields, com.mapbox.navigation.base.trip.model.RouteProgress routeProgress);
+    method public androidx.car.app.navigation.model.NavigationTemplate.NavigationInfo? mapNavigationInfo(com.mapbox.bindgen.Expected<com.mapbox.navigation.ui.maneuver.model.ManeuverError,java.util.List<com.mapbox.navigation.ui.maneuver.model.Maneuver>> expectedManeuvers, java.util.List<? extends com.mapbox.navigation.ui.shield.model.RouteShield> routeShields, com.mapbox.navigation.base.trip.model.RouteProgress routeProgress, com.mapbox.navigation.ui.maps.guidance.junction.model.JunctionValue? junctionValue);
```

Comment:
This would be a SEMVER if we were promising a stable api.

Do you want to move the `CarNavigationInfoMapper` into an internal directory so that it is still visible but changes will not break semver?

`package com.mapbox.androidauto.internal.navigation`

We can expose a public api for this is with a builder

``` kotlin
class CarNavigationInfo

 class Builder {
   fun build() = CarNavigationInfoMapper.map(param1, param2, param3)
}
```

Reply:
Are we already Stable? NAVAND-758

Reply:
No we're still releasing 0.x.0 versions. We have not released a `1.0.0`, I'm saying if we did we could not actually merge this pull request. So in order to have an api that supports changes better, we'll need to move `CarNavigationInfoMapper` out of public api

Reply:
We have run into issues with `JvmOverloads` because it creates a new function https://github.com/mapbox/mapbox-navigation-android/pull/6422#issuecomment-1263688121. But that issue is typically found by creating more than 2 sdk layers. (e.g., aa-sdk->nav-skd->map-sdk. map-sdk changes cause the crash)

May not be relevant here. but it's a reason to prefer builders over jvmoverloads.

JvmOverloads is better than nothing though, so feel free to update!